### PR TITLE
runtime: don't exit the main thread if a coexp attempted to gc it

### PR DESCRIPTION
**Pull request by Jafaral**

### CODE REVIEW

File: src/runtime/rmemmgt.r [hunk 1]
```diff
@@ -1535,6 +1535,8 @@ static void cofree()
 
          xep = *ep;
          *ep = (*ep)->nextstk;
+         if (xep->program == &rootpstate || xep->id == 1)
```
Comment (Don-Ward) [inline src/runtime/rmemmgt.r:1538]: This causes build errors for some configurations.

**Review by Don-Ward — CHANGES_REQUESTED** [review]
I think the headline comment is misleading. It says we should never GC the main thread, which is obviously wrong.  I _think_ the mod is attempting to stop the recovery of the activation block for the main thread (which makes sense, because nobody activated it). 

However, the big problem is that the ci fails. I put the patch into my (up to date) master branch on my local machine and it failed there too.

File: src/runtime/rcoexpr.r [hunk 2]
```diff
@@ -577,19 +577,25 @@ void coclean(struct b_coexpr *cp) {
...
+         * If the thread is cleaning itself, exit, what about tls chain?
+         */
+         cp->alive = -1;         /* signal thread to exit */
+         cp->have_thread = 0;
+   #ifndef NO_COEXPR_SEMAPHORE_FIX
+         if (cp->semp) {SEM_CLOSE(cp->semp); cp->semp = NULL;}
+   #endif                  /* NO_COEXPR_SEMAPHORE_FIX */
+         pthread_exit(0);
+      } else if (cp->id == 1 && cp->id == curtstate->c->id) {
+         /* FIXME: Main thread, should just return? */
```
Comment (Don-Ward) [inline src/runtime/rcoexpr.r:593]: Should this FIXME be here?  If yes, it's new code; why aren't we fixing it now?

**Review by Don-Ward — COMMENTED** [review]
Looks OK. I'm not sure if the FIXME describes what has been done (and should be removed) or if it is questioning the new code .

File: src/runtime/rcoexpr.r (hunk 2 above)
Comment (Jafaral) [reply src/runtime/rcoexpr.r:593]: Left it there intentionally because I wasn't there if this has any side effects.

**Pull request merged**